File: test_distance.R
source("distance.R")

cities <- list(
  list(name = "Belo Horizonte, Brasil", p = c(-19.9191, -43.9386)),
  list(name = "Sao Jose dos Campos, Brasil", p = c(-23.1848, -45.8784)),
  list(name = "Manaus, Brasil", p = c(-3.1301, -60.0234)),
  list(name = "Miami, EUA", p = c(25.7715, -80.1920)),
  list(name = "Toquio, Japao", p = c(35.6812, 139.7619))
)

cat("Distances between cities (geodesic vs euclidean) in kilometers:\n\n")

for (i in seq_len(length(cities) - 1)) {
  for (j in seq((i + 1), length(cities))) {
    name_city1 <- cities[[i]]$name
    name_city2 <- cities[[j]]$name
    p1 <- cities[[i]]$p
    p2 <- cities[[j]]$p
    D <- distance$new(p1, p2)
    geo <- D$geodesic()
    euc <- D$euclidean()
    cat(sprintf(
      "%s <-> %s: geodesic=%.2f km, euclidean=%.2f km\n",
      name_city1,
      name_city2,
      geo,
      euc
    ))
  }
}

File: test_distance.py
from distance import distance

cities = [
    ("Belo Horizonte, Brasil", (-19.9191, -43.9386)),
    ("Sao Jose dos Campos, Brasil", (-23.1848, -45.8784)),
    ("Manaus, Brasil", (-3.1301, -60.0234)),
    ("Miami, EUA", (25.7715, -80.1920)),
    ("Toquio, Japao", (35.6812, 139.7619)),
]


def main():
    print("Distances between cities (geodesic vs euclidean) in kilometers:\n")
    for i in range(len(cities)):
        for j in range(i + 1, len(cities)):
            name_city1, p1 = cities[i]
            name_city2, p2 = cities[j]
            D = distance(p1, p2)
            geo = D.geodesic()
            euc = D.euclidean()
            print(f"{name_city1} <-> {name_city2}: geodesic={geo:.2f} km, euclidean={euc:.2f} km")


if __name__ == "__main__":
    main()
